crypto/src/fiat_shamir/default_transcript.rs

```diff
@@ -62,7 +62,11 @@ where
     }
 
     fn sample_field_element(&mut self) -> FieldElement<F> {
-        FieldElement::from_bytes_be(&self.sample()).unwrap()
+        loop {
+            if let Some(result) = F::get_random_field_element_from_seed(self.sample()) {
+                return result;
+            }
+        }
```

Comment:
Following from Idea 2, this could become:
```suggestion
        let rng = <rand_chacha::ChaChaXRng as SeedableRng>::from_seed(self.sample);
        F::get_random_field_element_from_rng(rng)
```
where `X` in `ChaChaX` is 8, 12, or 20 for your favourite security level :)